src/Storages/MergeTree/MutateTask.cpp

```diff
@@ -1343,6 +1393,17 @@ bool PartMergerWriter::iterateThroughAllProjections()
     return true;
 }
 
+bool PartMergerWriter::iterateThroughAllProjectionsToMask()
+{
+    for (const auto & mutation_context : ctx->projection_mutation_contexts)
+    {
+        Block cur_block;
+        while (mutation_context->mutating_executor->pull(cur_block))
+            mutation_context->out->write(cur_block);
```

Comment:
It breaks the logic of `MutateTask` which is designed to be executed in small steps, because it executes the whole mutation in one step.

Reply:
solved